libenv/sysinfo.c

```diff
@@ -2311,6 +2317,31 @@ static int Linux_Mandriva_Version_Real(EvalContext *ctx, char *filename, char *r
 
 /******************************************************************/
 
+static int Linux_Amazon_Version(EvalContext *ctx)
+{
+    char buffer[CF_BUFSIZE];
+
+    // Amazon Linux AMI release 2016.09
+
+    if (ReadLine("/etc/system-release", buffer, sizeof(buffer)))
```

Comment:
`/etc/os-release` does exist but right now it just parsing a single line from `/etc/system-release` is the easiest way to determine if the host os is Amazon Linux.

At some point it seems like a lot of the logic in sysinfo.c can be refactored to take advantage of `/etc/os-release` but even this is not that simple since different distributions take liberties with these parameters. (For instance, on a debian 8 machine, neither VERSION_ID or VERSION tell you what point release it is so you would still need to parse /etc/debian_version to get this info)